pkg/reactor/download_controller.go

```diff
@@ -110,14 +111,7 @@ func (d *_downloadWorker) Work(ctx context.Context, ctrl *downloadController, ta
 	if task, ok := task.(*DownloadTask); ok {
 		if !ctrl.isDownloaded(task) {
 			if err := d.download(ctx, task); err != nil {
-				// find out any other recorded references that will fail
-				var refs string
-				if tasks, ok := ctrl.downloadedResources[task.Source]; ok {
-					for _, t := range tasks {
-						refs = fmt.Sprintf("%sReference %s from referer %s\n", refs, t.Reference, t.Referer)
-					}
-				}
-				refs = fmt.Sprintf("%sReference %s from referer %s", refs, task.Reference, task.Referer)
```

Comment:
I agree this doesn't work as intended, plus it provides a racing condition. The intention was that we report back all references to a failed download packed together instead of scattered in the log. 
My suggestion is at minimum to fix the racing condition in this PR (as you did) and look for readability/analysis improvements if necessary in future.

Reply:
Also, not sure if lines 119-120 are really necessary.